controller/upload_file.rst

```diff
@@ -442,10 +442,16 @@ controller.
             public function postLoad(LifecycleEventArgs $args)
             {
                 $entity = $args->getEntity();
-
-                $fileName = $entity->getBrochure();
-
-                $entity->setBrochure(new File($this->targetPath.'/'.$fileName));
+                
+                if (!$entity instanceof Product) {
+                    return;
+                }
+
+                if ($entity->getBrochure()) {
+                    $fileName = $entity->getBrochure();
```

Comment:
Oh, see want you thinking about, in this case, yes, that work but let's imagine we have an array of files, with this approach, the array must be read before the instantiation of File, let's imagine the case of an article with multiples images linked : 

```php

    /**
     * @param LifecycleEventArgs $args
     *
     * @throws FileNotFoundException
     */
    public function postLoad(LifecycleEventArgs $args)
    {
        $entity = $args->getObject();

        if (!$entity instanceof Article) {
            return;
        }

        // Only if the entity has store files.
        if ($entity->getImages()) {
            $filename = $entity->getImages();

            foreach ($filename as $file) {
                $entity->addImage(new File($this->imagesDir.'/'.$file));
            }
        }
    }
``` 

In this particular case, we must read every every entry and check if the entity return any entry into the array, this way, $filename become an array of images and we can instantiate every images into a file, i know, that's a particular case but my vision was about thinking this type of case.